core/dbt/include/global_project/macros/materializations/models/materialized_view/materialized_view.sql

```diff
@@ -0,0 +1,74 @@
+{%- materialization materialized_view, default -%}
+  {% set full_refresh_mode = should_full_refresh() %}
+  {%- set existing_relation = load_cached_relation(this) -%}
+  {%- set target_relation = this.incorporate(type='view') -%}
+  {%- set intermediate_relation =  make_intermediate_relation(target_relation) -%}
+
+  -- the intermediate_relation should not already exist in the database; get_relation
+  -- will return None in that case. Otherwise, we get a relation that we can drop
+  -- later, before we try to use this name for the current operation
+  {%- set preexisting_intermediate_relation = load_cached_relation(intermediate_relation) -%}
+  /*
+     This relation (probably) doesn't exist yet. If it does exist, it's a leftover from
+     a previous run, and we're going to try to drop it immediately. At the end of this
+     materialization, we're going to rename the "existing_relation" to this identifier,
+     and then we're going to drop it. In order to make sure we run the correct one of:
+       - drop view ...
+       - drop table ...
+     We need to set the type of this relation to be the type of the existing_relation, if it exists,
+     or else "view" as a sane default if it does not. Note that if the existing_relation does not
+     exist, then there is nothing to move out of the way and subsequentally drop. In that case,
+     this relation will be effectively unused.
+  */
+  {%- set backup_relation_type = 'view' if existing_relation is none else existing_relation.type -%}
+  {%- set backup_relation = make_backup_relation(target_relation, backup_relation_type) -%}
+  -- as above, the backup_relation should not already exist
+  {%- set preexisting_backup_relation = load_cached_relation(backup_relation) -%}
+  -- grab current tables grants config for comparision later on
+  {% set grant_config = config.get('grants') %}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+  -- drop the temp relations if they exist already in the database
+  {{ drop_relation_if_exists(preexisting_intermediate_relation) }}
+  {{ drop_relation_if_exists(preexisting_backup_relation) }}
+
+  -- `BEGIN` happens here:
+  {{ run_hooks(pre_hooks, inside_transaction=True) }}
+
+
+  -- cleanup
+  -- move the existing view out of the way
+  {% if existing_relation is none %}
+      {% set build_sql = create_materialized_view_as(target_relation, sql, config) %}
+
+  {% elif full_refresh_mode or existing_relation.type != 'materializedview' %}
+      {#-- Make sure the backup doesn't exist so we don't encounter issues with the rename below #}
```

Comment:
I think three lines are already handled with lines 24, 26, and 34 above. I would look at `intermediate.sql` and `table.sql` as well to see how those were implemented.

Also, we will need an intermediate object for the recreation because we'll have data in the existing object and we'll need to load the new object up; so there's a non-trivial amount of downtime.

Reply:
the entirety of the if statement or just specific part? both `table.sql` and `view.sql` have both a similar setup around the `backup_relation` setup, then the `build model` where we build out the main call statement and then this cleanup conditional. but for the MV would we not need to know if full refresh is triggered/dropped already before we build the sql_model?

Reply:
Sorry, could have been more clear. I was referring to lines 43-5. But it now looks like we're not dropping the backup relation above if it already exists, just the intermediate one. I would put all of that setup before this if block since it doesn't really cost much to try to drop something that doesn't exist; and we would want to make sure we're clean before doing work, regardless of the logic in the if block.

Reply:
ah, that makes sense will re-add and remove from the conditional